compiler/codegen/library.cpp

```diff
@@ -189,3 +196,163 @@ void closeLibraryHelperFile(fileinfo* fi, bool beautifyIt) {
   if (beautifyIt)
     beautify(fi);
 }
+
+static void pxdEnd() {
+  FILE* pxd = gGenInfo->cfile;
+
+  // Do not "clean up" this file, as it removes leading tabs and those are
+  // necessary.
+  fprintf(pxd, "\ncdef extern from \"chpltypes.h\":\n");
+  fprintf(pxd, "\tctypedef void* c_fn_ptr\n\n");
+
+  fprintf(pxd, "cdef extern from \"chpl-init.h\":\n");
+  fprintf(pxd, "\tvoid chpl_library_init(int argc, char* argv[])\n");
+  fprintf(pxd, "\tvoid chpl_library_finalize()\n");
+}
+
+// Populate the pythonNames map with the translation for bools, differently sized
+// integers, etc.
+static void setupPythonTypeMap() {
+  pythonNames[dtInt[INT_SIZE_8]->symbol] = std::make_pair("", "numpy.int8");
+  pythonNames[dtInt[INT_SIZE_16]->symbol] = std::make_pair("", "numpy.int16");
+  pythonNames[dtInt[INT_SIZE_32]->symbol] = std::make_pair("", "numpy.int32");
+  pythonNames[dtInt[INT_SIZE_64]->symbol] = std::make_pair("", "numpy.int64");
+  pythonNames[dtUInt[INT_SIZE_8]->symbol] = std::make_pair("", "numpy.uint8");
+  pythonNames[dtUInt[INT_SIZE_16]->symbol] = std::make_pair("", "numpy.uint16");
+  pythonNames[dtUInt[INT_SIZE_32]->symbol] = std::make_pair("", "numpy.uint32");
+  pythonNames[dtUInt[INT_SIZE_64]->symbol] = std::make_pair("", "numpy.uint64");
+  pythonNames[dtReal[FLOAT_SIZE_32]->symbol] = std::make_pair("", "numpy.float32");
+  pythonNames[dtReal[FLOAT_SIZE_64]->symbol] = std::make_pair("double", "float");
+  pythonNames[dtBool->symbol] = std::make_pair("bint", "bint");
+  pythonNames[dtStringC->symbol] = std::make_pair("char *", "bytes");
+  pythonNames[dtComplex[COMPLEX_SIZE_64]->symbol] =
+              std::make_pair("float complex", "numpy.complex64");
+  pythonNames[dtComplex[COMPLEX_SIZE_128]->symbol] =
+              std::make_pair("double complex", "numpy.complex128");
+
+  /* TODO: // Handle bigint
+  forv_Vec(TypeSymbol, t, gTypeSymbols) {
+    if (
+  }
+  */
+
+}
+
+// If there is a known .pxd file translation for this type, use that.
+// Otherwise, use the normal cname
+static std::string getPXDTypeName(Type* type) {
+  std::pair<std::string, std::string> tNames = pythonNames[type->symbol];
+  if (tNames.first != "") {
+    return tNames.first;
+  } else {
+    return transformTypeForPointer(type);
+  }
+}
+
+std::string ArgSymbol::getPXDType() {
```

Comment:
I think we have tried to define codegen methods in codegen/symbol.cpp - do you think that's practical in this case?

Reply:
It's certainly doable.  I was leaning towards keeping the related functionality in the same place so it would be easier to find it all, but having all the ArgSymbol functions in the same place for codegen also is reasonable